File: csharp/MapleStory.OpenAPI/Src/Dto/Character/CharacterBasicDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 기본 정보
    /// </summary>
    public class CharacterBasicDTO
    {
        /// <summary>
        /// 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
        /// </summary>
        [JsonProperty("date")]
        public DateTimeOffset? Date
        {
            get
            {
                return _date?.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _date = value;
            }
        }

        private DateTimeOffset? _date;

        /// <summary>
        /// 캐릭터 명
        /// </summary>
        [JsonProperty("character_name")]
        public string CharacterName { get; set; }

        /// <summary>
        /// 월드 명
        /// </summary>
        [JsonProperty("world_name")]
        public string WorldName { get; set; }

        /// <summary>
        /// 캐릭터 성별
        /// </summary>
        [JsonProperty("character_gender")]
        public string CharacterGender { get; set; }

        /// <summary>
        /// 캐릭터 직업
        /// </summary>
        [JsonProperty("character_class")]
        public string CharacterClass { get; set; }

        /// <summary>
        /// 캐릭터 전직 차수
        /// </summary>
        [JsonProperty("character_class_level")]
        public string CharacterClassLevel { get; set; }

        /// <summary>
        /// 캐릭터 레벨
        /// </summary>
        [JsonProperty("character_level")]
        public long CharacterLevel { get; set; }

        /// <summary>
        /// 현재 레벨에서 보유한 경험치
        /// </summary>
        [JsonProperty("character_exp")]
        public long CharacterExp { get; set; }

        /// <summary>
        /// 현재 레벨에서 경험치 퍼센트
        /// </summary>
        [JsonProperty("character_exp_rate")]
        public string CharacterExpRate { get; set; }

        /// <summary>
        /// 캐릭터 소속 길드 명
        /// </summary>
        [JsonProperty("character_guild_name")]
        public string? CharacterGuildName { get; set; }

        /// <summary>
        /// 캐릭터 외형 이미지
        /// </summary>
        [JsonProperty("character_image")]
        public string CharacterImage { get; set; }

        /// <summary>
        /// 캐릭터 생성일
        /// </summary>
        [JsonProperty("character_date_create")]
        public DateTimeOffset CharacterDateCreate
        {
            get
            {
                return _characterDateCreate.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _characterDateCreate = value;
            }
        }

        private DateTimeOffset _characterDateCreate;

        /// <summary>
        /// 최근 7일간 접속 여부
        /// </summary>
        public bool AccessFlag
        {
            get
            {
                return _accessFlag == "true";
            }
        }

        [JsonProperty("access_flag")]
        private string _accessFlag { get; set; }

        /// <summary>
        /// 해방 퀘스트 완료 여부
        /// </summary>
        public bool LiberationQuestClearFlag
        {
            get
            {
                return _liberationQuestClearFlag == "true";
            }
        }

        [JsonProperty("liberation_quest_clear_flag")]
        private string _liberationQuestClearFlag { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/character/CharacterBasicDTO.java
package dev.spiralmoon.maplestory.api.dto.character;

import com.google.gson.annotations.SerializedName;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.time.LocalDateTime;

/**
 * 캐릭터 기본 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterBasicDTO {

    /**
     * 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
     */
    @SerializedName("date")
    private String date;

    /**
     * 캐릭터 명
     */
    @SerializedName("character_name")
    private String characterName;

    /**
     * 월드 명
     */
    @SerializedName("world_name")
    private String worldName;

    /**
     * 캐릭터 성별
     */
    @SerializedName("character_gender")
    private String characterGender;

    /**
     * 캐릭터 직업
     */
    @SerializedName("character_class")
    private String characterClass;

    /**
     * 캐릭터 전직 차수
     */
    @SerializedName("character_class_level")
    private String characterClassLevel;

    /**
     * 캐릭터 레벨
     */
    @SerializedName("character_level")
    private long characterLevel;

    /**
     * 현재 레벨에서 보유한 경험치
     */
    @SerializedName("character_exp")
    private long characterExp;

    /**
     * 현재 레벨에서 경험치 퍼센트
     */
    @SerializedName("character_exp_rate")
    private String characterExpRate;

    /**
     * 캐릭터 소속 길드 명
     */
    @SerializedName("character_guild_name")
    private String characterGuildName;

    /**
     * 캐릭터 외형 이미지
     */
    @SerializedName("character_image")
    private String characterImage;

    /**
     * 캐릭터 생성일
     */
    @SerializedName("character_date_create")
    private String characterDateCreate;

    /**
     * 최근 7일간 접속 여부
     */
    @SerializedName("access_flag")
    private String accessFlag;

    /**
     * 해방 퀘스트 완료 여부
     */
    @SerializedName("liberation_quest_clear_flag")
    private String liberationQuestClearFlag;

    /**
     * 조회 기준일
     */
    public LocalDateTime getDate() {
        return date != null
                ? Utils.toLocalDateTime(this.date)
                : null;
    }

    /**
     * 캐릭터 생성일
     */
    public LocalDateTime getCharacterDateCreate() {
        return Utils.toLocalDateTime(this.characterDateCreate);
    }

    /**
     * 최근 7일간 접속 여부
     */
    public boolean isAccessFlag() {
        return "true".equals(this.accessFlag);
    }

    /**
     * 해방 퀘스트 완료 여부
     */
    public boolean isLiberationQuestClearFlag() {
        return "true".equals(this.liberationQuestClearFlag);
    }
}
